src/app/candidate-management/candidate-management-detail/candidate-management-detail.component.ts

```diff
@@ -60,47 +62,66 @@ export class CandidateManagementDetailComponent implements OnInit, AfterViewInit
-        this.contentToMarkdown();
-        this.checkTreshhold();
-
-      } else if (_candidate && this.router.url.includes('edit')) {
-        this.candidate = _candidate;
-        this.contentToMarkdown();
-        this.edit();
-        this.checkTreshhold();
-
-      } else if (!_candidate && window.history.state.data && window.history.state.data instanceof Candidate) {
-        this.candidate = window.history.state.data as Candidate
-        this.contentToMarkdown();
-        this.edit();
-        this.checkTreshhold();
-
-      } else {
-        this.disabled = false;
-        this.candidate = new Candidate(null, 'New Candidate', null, null);
-        this.patternLanguageSelectedChange(patternLanguageNone);
-        // Preset author
-        this.auth.user.subscribe(_user => {
-          if (_user && !this.candidate.authors) this.candidate.authors = [new AuthorModel(_user.id, Author.OWNER, _user.name)];
-        })
-      }
-      this.confirmDialog = {
-        title: `Change Pattern Language for Candidate ${this.candidate.name}`,
-        text: 'If you change the language everything writen will be deleted and the'
-          + ' new pattern schema will be used'
+    this.arSubscription = this.activeRoute.params.subscribe(params => {
+      let candidateUri = `/candidates/${params.name}`;
+      switch (params.action) {
+        case 'detail': {
+          this.disabled = true;
+          this.candidateManagementService.getCandidateByUri(candidateUri).subscribe(result => {
+            this.candidate = result;
+            this.contentToMarkdown();
+            this.checkTreshhold();
+            this.setupPLChangeDialog();
+          });
+          break;
+        }
+        case 'edit': {
+          this.candidateManagementService.getCandidateByUri(candidateUri).subscribe(result => {
+            this.candidate = result;
+            this.contentToMarkdown();
+            this.edit();
+            this.checkTreshhold();
+            this.setupPLChangeDialog();
+          });
+          break;
+        }
+        case 'create': {
+          this.disabled = false;
+          let candidateName = params.name ? params.name : 'New Candidate';
+          this.candidate = new Candidate(null, candidateName, null, null);
+          this.patternLanguageSelectedChange(patternLanguageNone);
+          // Preset author
+          this.auth.user.subscribe(_user => {
+            if (_user && !this.candidate.authors) this.candidate.authors = [new AuthorModel(_user.id, Author.OWNER, _user.name)];
+          });
+          this.setupPLChangeDialog();
+          break;
+        }
+        default: {
+          // Unknown action - show candidate list
+          this.router.navigateByUrl('/candidate');
+          break;
+        }
       }
     });
   }
 
+  ngOnDestroy(): void {
+    this.arSubscription.unsubscribe();
+  }
+
   ngAfterViewInit(): void {
     this.setCommentSectionHeight();
   }
 
+  setupPLChangeDialog() {
```

Comment:
Refactor this into a getter (or a get... function) that calculates the dialog title & text from the current candidate (maybe even from a parameter and not from this).
Currently one must remember to call this function on every update on the candidate.